include/ziapi/Http.hpp

```diff
@@ -59,7 +59,7 @@ class IResponseInputQueue {
 public:
     virtual ~IResponseInputQueue() = default;
 
-    virtual std::pair<Request, Context> Pop() = 0;
+    virtual bool Pop(std::pair<Request, Context> &req) = 0;
```

Comment:
Hi, why not use structure binding to return the result as well as the boolean describing the state, it would be cleaner as the purpose of Pop is to remove the last element and return it.

```suggestion
    virtual std::pair<std::pair<Request, Context>, bool> Pop() = 0;
```